spec/classes/nfs_spec.rb

```diff
@@ -1,187 +1,244 @@
 require 'spec_helper'
 
-describe 'nfs', :type => 'class' do
-  context "server => true" do
+describe 'nfs', type: 'class' do
+  context 'server => true' do
     let :params do {
-      :server_enabled => true,
-      :client_enabled => false,
-    } end
+      server_enabled: true,
+      client_enabled: false,
+    }
+    end
     let(:facts) { {
-      :operatingsystem => 'Ubuntu',
-      :osfamily => 'Debian',
-      :operatingsystemmajrelease => '12.04',
-      :concat_basedir => '/tmp',
-      :is_pe => false,
-      :id => 'root',
-      :path => '/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin',
+      operatingsystem: 'Ubuntu',
+      osfamily: 'Debian',
+      operatingsystemmajrelease: '12.04',
+      concat_basedir: '/tmp',
+      is_pe: false,
+      id: 'root',
+      path: '/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin',
```

Comment:
this whole spec test is a profoundly confusing mess of repetition and redundancies with only very slight variations on the theme.
go radical, or go home: replace the facts with the os facts gem.